### ISSUE TITLE
Config File Name

### ISSUE THREAD

**Issue by paddatrapper** [libretime/libretime#420]
We should rename the configuration file to `libretime.ini` and expect it in `/etc/libretime/`. Backwards compatibility could be provided by symlinking `airtime.ini`in the install script

**Comment by hairmare** [comment]
I'm +1 for this but would like to do it after having tagged 3.0.0-alpha.4 since it's a change that needs to get some git master testing before we can release it. I'll tag it soon as i believe we got the uploading problems from this release window (#352) somewhat pinned down to have info on them in the release notes to soften the blow issue wise.

IMO backwards compat isn't strictly needed as all the code bits that use the file already use env variable to get the actual files location so changing the file location is basically config at this point.

**Comment by paddatrapper** [comment]
Ah Ok, yeah env variables does make it easier. Installer issues being where its installed and setting the env variable?

**Comment by hairmare** [comment]
I was under the impression that its cleaner but turned out to be wrong :(

Some parts are okayish

https://github.com/LibreTime/libretime/blob/6e2cb2b2a80442f5a5aa9e181649393bdabd6ea7/installer/systemd/airtime-celery.service#L8-L10

PHP has the naming in one central location

https://github.com/LibreTime/libretime/blob/4557395a865ed8d7592604b73c47ed3f5353c9dc/airtime_mvc/public/index.php#L34-L36

And then there is also heaps of legacy thingies

https://github.com/LibreTime/libretime/blob/276a69aa1bef527260efc170d7caab7d04b4ca13/python_apps/pypo/liquidsoap/liquidsoap_prepare_terminate.py#L5-L7

Since we need to touch the code anyway, lets somewhat standardize this. Overhauling config reading has been on my wishlist for a while anyway.

I'm not sure how far we want to go, using yaml instead of ini would make lots of the php python compat problems we have today go away if implemented properly.

If we keep the ini file we could read the libretime.ini location first and issue a warning if it doesn't exit and then try reading airtime.ini.

For the python parts it might make sense to implement config loading in a common module for the individual apps to use. The php parts already have the config file reading in one location so it's fairly easy to integrate there.

**Comment by paddatrapper** [comment]
I'm definitely with you with overhauling the config stuff. Yaml would be nice, but developer availability is an issue I guess... Unfortunately I'm reluctant to put my hand up for it because life will get crazy soon and I've still got a fair amount of packaging to do.

I'm not sure how much backwards compatibility we want to maintain. Perhaps making a separate script for converting from airtime to libretime configs will be useful if we break compatibility

Yeah, common module is good. There are a couple good yaml modules and built in ini support in python

**Comment by hairmare** [comment]
I'm not sure when I'll find time too look into the config parts. It is certainly a royal PITA. The thing with the python ini module is it being opinionated about the file syntax in a very different way than PHPs parse_ini_file function.

I've been planning on starting with parts of the python overhaul (ie. using future to get ready for py3) soonish and that should at least have a positive effect on making the ini file name changeable.

Let's keep this about the rename and create a new issue when we want to tackle any config file changes that go further.

**Comment by paddatrapper** [comment]
I'm going to look into renaming the config file over the next week. I will leave any config file format changes for later, but I will try consolidate the config loading in PHP and Python

**Comment by jooola** [comment]
Hey, I would like to tackle this one. I guess no body assigned himself to this task ?

Do we still want to use environment variables ? I think it's a great idea.

In order to read the configuration from an app:
should the app directly target `os.environ` to get the entries, and simply do a validation on application startup.
Or
Should we wrap the environment variable inside a dictionary the python apps can retrieve, with validation.

I don't know if we rely on some config hot reload, maybe that's useful ?

This integrates well with sytsemd and docker. For systemd a environment variables [can be loaded from service file or env file](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Environment), for docker its [works almost the same](https://docs.docker.com/compose/environment-variables/). Don't know about the other technologies.

This change is in multiple steps:
First: 
- Rename everything to libretime (Why not a project wide search and replace ? Why has nobody done this yet ?)
- Add migration script to move old files to new location and provide symlinks for old location to new.

Then:
- Create config management utility for python apps
- Migrate settings to use this config manager
- Change php config reading.

**Comment by paddatrapper** [comment]
> Hey, I would like to tackle this one. I guess no body assigned himself to this task ?

Sure, see some work in https://github.com/paddatrapper/libretime/tree/config-file where I started on it, but ran out of time.

> Do we still want to use environment variables ? I think it's a great idea.

Yes, with a default to `/etc/libretime/libretime.conf`

> I don't know if we rely on some config hot reload, maybe that's useful ?

We don't and probably don't want to - Django in particular is fussy about settings changes.

> Why not a project wide search and replace ? Why has nobody done this yet ?

Time, effort, interest, backwards compatibility, take your pick... There has been some move towards it, but the rest still needs doing. There could be other potential issues with a global rename that CI may fail to catch.

> Add migration script to move old files to new location and provide symlinks for old location to new.

This would need to support migration from Airtime to LibreTime (#71) as well

> Create config management utility for python apps

Note that the API uses the config too, just differently to the rest of the python applications so as to tie into Django correctly.
